integrationtests/test_i18n.py

```diff
@@ -20,3 +53,88 @@ def test_js_english_interpolation(self):
         # Wait for page to load
         b.assert_element('input[name="name"][value="Untitled Workflow"]', wait=True)
         b.assert_element(".attribution .metadata", text="by Jane Doe")
+
+
+class TestLocaleSwitch(LocaleSwitcherIntegrationTest):
+    def setUp(self):
+        super().setUp()
+        self.user = self.account_admin.create_user(
+            "user@example.org", first_name="Jane", last_name="Doe"
+        )
+
+    def test_language_switch_in_django_templates(self):
+        # You can change from "en" to "el" and back and the UI will be translated
+        b = self.browser
+        b.visit("/")
+        b.assert_element('input[name="password"][placeholder="Password"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Sign In")
+        self.switch_locale_django("el")
+        b.assert_element('input[name="password"][placeholder="Συνθηματικό"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Σύνδεση")
+        self.switch_locale_django("en")
+        b.assert_element('input[name="password"][placeholder="Password"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Sign In")
+
+    def test_language_switch_in_react(self):
+        # You can change from "en" to "el" and back and the UI will be translated
+        b = self.browser
+
+        accounts.login(b, self.user.email, self.user.email)
+        b.visit("/workflows/")
+
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Create Workflow",
+            wait=True,
+        )
+        self.switch_locale_react("en", "el")
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Δημιουργία ροής εργασιών",
+            wait=True,
+        )
```

Comment:
Before switching locales, navigate to another page (`b.click("[Greek for 'Lessons']")` springs to mind) to check that the chosen language persists to the next page.